impeller/core/device_buffer.cc

```diff
@@ -69,4 +70,11 @@ const DeviceBufferDescriptor& DeviceBuffer::GetDeviceBufferDescriptor() const {
   return OnCopyHostBuffer(source, source_range, offset);
 }
 
+std::unique_ptr<const uint8_t[]> DeviceBuffer::CopyToUnqiueBuffer() const {
+  auto size = GetDeviceBufferDescriptor().size;
+  auto raw_data = static_cast<uint8_t*>(malloc(size));
+  memcpy(raw_data, OnGetContents(), size);
+  return std::unique_ptr<const uint8_t[]>(raw_data);
```

Comment:
This is leaking right now because I need to give it a release proc

Reply:
Rather than lugging around a device allocation, you could just eat an extra host copy now and we can repair this later when we [fix up the blits](https://github.com/flutter/engine/pull/46394#discussion_r1341864534).